account_payment_term_restriction/models/account_payment_term.py

```diff
@@ -0,0 +1,61 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class AccountPaymentTerm(models.Model):
+    _inherit = "account.payment.term"
+
+    applicable_on = fields.Selection(
+        selection="_selection_applicable_on",
+        string="Applicable On",
+        default=lambda self: self._get_default_applicable_on(),
+    )
+
+    @api.model
+    def _selection_applicable_on(self):
+        return [
+            ("sale", "Sales"),
+            ("purchase", "Purchases"),
+            ("all", "All"),
+        ]
+
+    @api.model
+    def _get_default_applicable_on(self):
+        return "all"
+
+    @api.model
+    def get_sale_applicable_on(self):
+        return ["sale", "all"]
+
+    @api.model
+    def get_purchase_applicable_on(self):
+        return ["purchase", "all"]
+
+    def _check_not_applicable_correct_values(self, record):
+        return (
+            self
+            and len(self) == 1
```

Comment:
you can use self.ensure_one() at the beginning

Reply:
self.ensure_one() would throw an exception, in my case I just want the function to return False if the pre-checks are not fulfilled.